frontend/src/app/person-network-members/person-network-members.component.ts

```diff
@@ -0,0 +1,74 @@
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+import { NetworkMemberModel } from '../models/network-member.model';
+import { ConfirmService } from '../confirm.service';
+import { NetworkMemberService } from '../network-member.service';
+import { switchMap, tap } from 'rxjs/operators';
+import { PersonModel } from '../models/person.model';
+import { NETWORK_MEMBER_TYPE_TRANSLATIONS } from '../display-network-member-type.pipe';
+import { FormBuilder, FormGroup, Validators } from '@angular/forms';
+import { NetworkMemberCommand } from '../models/network-member.command';
+import { Observable } from 'rxjs';
+
+@Component({
+  selector: 'gl-person-network-members',
+  templateUrl: './person-network-members.component.html',
+  styleUrls: ['./person-network-members.component.scss']
+})
+export class PersonNetworkMembersComponent implements OnInit {
+
+  members: Array<NetworkMemberModel>;
+  person: PersonModel;
+  memberTypes = NETWORK_MEMBER_TYPE_TRANSLATIONS.map(t => t.key);
+
+  editedMember: NetworkMemberModel;
+  memberForm: FormGroup;
+
+  constructor(private route: ActivatedRoute,
+              private confirmService: ConfirmService,
+              private networkMemberService: NetworkMemberService,
+              private fb: FormBuilder) { }
+
+  ngOnInit() {
+    this.members = this.route.snapshot.data.members;
+    this.person = this.route.snapshot.parent.data.person;
+  }
+
+  delete(member: NetworkMemberModel) {
+    this.confirmService.confirm({
+      message: 'Voulez-vous vraiment supprimer ce membre de réseau\u00a0?'
+    }).pipe(
+      switchMap(() => this.networkMemberService.delete(this.person.id, member.id)),
+      switchMap(() => this.networkMemberService.list(this.person.id)),
+    ).subscribe(members => this.members = members, () => {});
```

Comment:
Is the empty error handler necessary?

Reply:
Yes. I chose to emit an error when the use doesn't confirm. If I don't catch that error, the test testing what happens when the user doesn't confirm blows up, and the error also appears in the console, which is ugly.

I could have chosen to emit true/false, but then I'd have to remember to filter true values every time I use confirm, which is worse, IMO.